js/showAllSongs.js

```diff
@@ -20,40 +21,91 @@ var comparator = function(property) {
   };
 };
 
-window.onload = function() {
+const sortSongsByArtist = function(songs) {
   const allSongs = {};
+  songs.map(function(song) {
+    // Sorting by artist
+    const artist = song["artist"];
+
+    if (allSongs.hasOwnProperty(artist)) {
+      allSongs[artist].push(song);
+    } else {
+      allSongs[artist] = [song];
+    }
+  });
+
+  let allSongsByArtist = [];
+  for (let artist in allSongs) {
+    // Song titles in alphabetical order
+    const songsByArtist = allSongs[artist].sort(comparator("title"));
+    allSongsByArtist.push({ "artist": artist, "songs": songsByArtist });
+  }
+  // Artists in alphabetical order
+  allSongsByArtist = allSongsByArtist.sort(comparator("artist"));
+  return allSongsByArtist;
+};
+
+const ALL_TAG = "all";
+let selectedTag = ALL_TAG;
+const allSongsByTag = {};
+
+const renderSelectedSongs = function(selectedTag) {
+  const songsToRender = allSongsByTag[selectedTag];
+  const sortedSongs = sortSongsByArtist(songsToRender);
+
+  document.getElementById("allSongs").innerHTML = allSongsListTemplate({
+    allSongs: sortedSongs,
+  });
+};
 
+window.onload = function() {
   $.ajax({
     url: "/static/data/ALL_SONGS.json",
     dataType: "json",
     success: function(data) {
-      data.map(function(song) {
+      const allTags = new Set();
+      const allSongs = [];
 
-        // Sorting by artist
-        const artist = song["artist"];
+      allTags.add(ALL_TAG);
+      data.map(function(song) {
+        allSongs.push(song);
 
-        if (allSongs.hasOwnProperty(artist)) {
-          allSongs[artist].push(song);
-        } else {
-          allSongs[artist] = [song];
-        }
+        const tags = song["tags"];
+        tags.map((tag) => {
+          allTags.add(tag);
+          if (allSongsByTag.hasOwnProperty(tag)) {
+            allSongsByTag[tag].push(song);
+          } else {
+            allSongsByTag[tag] = [song];
+          }
+        });
       });
+      allSongsByTag[ALL_TAG] = allSongs;
 
-      let allSongsByArtist = [];
-      for (let artist in allSongs) {
-        // Song titles in alphabetical order
-        const songsByArtist = allSongs[artist].sort(comparator("title"));
-        allSongsByArtist.push({ "artist": artist, "songs": songsByArtist });
-      }
-      // Artists in alphabetical order
-      allSongsByArtist = allSongsByArtist.sort(comparator("artist"));
+      const allTagButtons = Array.from(allTags).map((tag) => {
+        return {
+          type: "tag",
+          name: tag,
+          value: tag
+        };
+      });
+      document.getElementById("allTags").innerHTML = buttonTemplate({
+        buttons: allTagButtons
+      });
```

Comment:
We should mark the `all` button as initially selected (right now no buttons are initially selected).